bson/encoder_example_test.go

```diff
@@ -121,3 +122,126 @@ func ExampleEncoder_UseJSONStructTags() {
 	fmt.Println(bson.Raw(buf.Bytes()).String())
 	// Output: {"name": "Cereal Rounds","sku": "AB12345","price_cents": {"$numberLong":"399"}}
 }
+
+func ExampleEncoder_multipleBSONDocuments() {
+	// Create an Encoder that writes BSON values to a bytes.Buffer.
+	buf := new(bytes.Buffer)
+	vw, err := bsonrw.NewBSONValueWriter(buf)
+	if err != nil {
+		panic(err)
+	}
+	encoder, err := bson.NewEncoder(vw)
+	if err != nil {
+		panic(err)
+	}
+
+	type Coordinate struct {
+		X int
+		Y int
+	}
+
+	// Use the encoder to marshal 5 Coordinate values as a sequence of BSON
+	// documents.
+	for i := 0; i < 5; i++ {
+		err := encoder.Encode(Coordinate{
+			X: i,
+			Y: i,
+		})
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Read each marshaled BSON document from the buffer and print them as
+	// Extended JSON by converting them to bson.Raw.
+	for {
+		doc, err := bson.ReadDocument(buf)
+		if err == io.EOF {
+			return
+		}
+		if err != nil {
+			panic(err)
+		}
+		fmt.Println(bson.Raw(doc).String())
```

Comment:
Why use "fmt.Println" in some examples and "fmt.Printf" in others?

Reply:
Some of the examples require a specific formatting verb to make more sense, like `%+v` to print the field names in a struct. Some don't require any special formatting verb and can just print the raw string.